python/paddle/distributed/auto_parallel/converter.py

```diff
@@ -0,0 +1,449 @@
+# limitations under the License.
+
+import paddle
+import warnings
+import logging
+import numpy as np
+from ..utils import get_logger
+
+
+class Converter(object):
+    """
+    Converter is a class object for auto parallel to convert a parameter from 
+    one parallel strategy to another one. Parameters will merge and slice value 
+    with their strategy when strategies are different.
+    """
+
+    def __init__(self, params_dict, pre_strategy, cur_strategy):
+        """
+        Args:
+            params_dict(dict): parameters' value of all ranks that to be converted. 
+                key is param's name(str), value is all ranks' data(list(numpy.ndarray))
+            pre_strategy(dict): parameters' distributed attribute of last training process.
+                key is param's name(str), value is param's distributed attribute in last 
+                training process.
+            cur_strategy(dict): parameters' distributed attribute of current rank.
+                key is param's name(str), value is param's distributed attribute in current
+                rank.
+        """
+        self._params_dict = params_dict
+        self._pre_strategy = pre_strategy
+        self._cur_strategy = cur_strategy
+        self._logger = get_logger(logging.INFO)
+
+    @property
+    def params_dict(self):
+        return self._params_dict
+
+    @property
+    def pre_strategy(self):
+        return self._pre_strategy
+
+    @property
+    def cur_strategy(self):
+        return self._cur_strategy
+
+    @params_dict.setter
+    def params_dict(self, params_dict):
+        if not params_dict:
+            raise ValueError("'params_dict' is None, "
+                             "the parameters to be converted cannot be None.")
+        if not isinstance(params_dict, dict):
+            raise TypeError(
+                "The type of 'params_dict' should be 'dict', but got '{}'.".
+                format(str(type(params_dict))))
+        self._params_dict = params_dict
+
+    @pre_strategy.setter
+    def pre_strategy(self, pre_strategy):
+        if not pre_strategy:
+            raise ValueError("'pre_strategy' is None, "
+                             "there are not parameter in pre process.")
+        if not isinstance(pre_strategy, dict):
+            raise TypeError("The type of 'pre_strategy' should be 'dict', "
+                            "but got '{}'.".format(str(type(pre_strategy))))
+        self._pre_strategy = pre_strategy
+
+    @cur_strategy.setter
+    def cur_strategy(self, cur_strategy):
+        if not cur_strategy:
+            warnings.warn("'cur_strategy' is None, "
+                          "there are not parameter in cur process")
+        if not isinstance(cur_strategy, dict):
+            raise TypeError("The type of 'cur_strategy' should be 'dict', "
+                            "but got '{}'.".format(str(type(cur_strategy))))
+        self._cur_strategy = cur_strategy
+
+    def convert(self, strict=True):
```

Comment:
explain meaning of "strict" argument in doc

Reply:
done.